### PR TITLE
Codemirror 3

### PR DESCRIPTION
**Pull request by Pomax**
switched over all the code to make use of codemirror3 rather than codemirror2

### CODE REVIEW

File: js/fc/ui/gutter-pointer.js [hunk 1]
```diff
@@ -30,23 +30,23 @@
...
     var highlight = $(".CodeMirror-gutter-text ." + highlightClass, wrapper);
     var svg = document.createElementNS(SVG_NS, "svg");
     var pointer = document.createElementNS(SVG_NS, "polygon");
-    var w = ($(".CodeMirror-gutter", wrapper).outerWidth() -
-             highlight.width()) * 2;
-    var h = highlight[0].getBoundingClientRect().height;
-    var pos = highlight.position();
-    
-    pos.left += highlight.width();
+    var w = jQuery(".CodeMirror-gutters").width()/2;
```
Comment (Pomax) [inline js/fc/ui/gutter-pointer.js]: should be $(...)

File: examples/bare-towtruck.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html class="deployment-type-development">
+  <head>
+    <meta charset="utf-8">
+    <base target="_blank">
+
+    <title>Barebones Friendlycode Editor</title>
```
Comment (Pomax) [inline examples/bare-towtruck.html]: This should be retitled to a "with towtruck" thing

File: js/fc/ui/gutter-pointer.js [hunk 3]
```diff
@@ -30,23 +30,23 @@
...
+    var collection = $(".gutter-mark."+highlightClass);
+    var first = collection.first()[0];
+    var last = collection.last()[0];
+    var h = last.getBoundingClientRect().bottom - first.getBoundingClientRect().top;
```
Comment (ScottDowne) [inline js/fc/ui/gutter-pointer.js]: You could probably make these all one var delcaration if you prefer.

var thing,
       thing2,
       etc;

Comment (Pomax) [inline js/fc/ui/gutter-pointer.js]: might as well.

**Pull request merged**